source/common/ssl/connection_impl.cc

```diff
@@ -251,8 +251,21 @@ std::string ConnectionImpl::subjectPeerCertificate() {
   if (!cert) {
     return "";
   }
-  bssl::UniquePtr<char> buf(X509_NAME_oneline(X509_get_subject_name(cert.get()), nullptr, 0));
-  return std::string(buf.get());
+
+  // Format settings for a single line in RFC 2253 format.
+  constexpr int indent = 0;
+  constexpr unsigned long flags = XN_FLAG_RFC2253;
```

Comment:
RFC2253 says that:

    When converting from an ASN.1 RelativeDistinguishedName to a string,
    the output consists of the string encodings of each
    AttributeTypeAndValue (according to 2.3), in any order.

So I both orders are fine (`XN_FLAG_RFC2253` is printed in reverse order).

In fact, both: `XN_FLAG_RFC2253` and `XN_FLAG_ONELINE` are using RFC2253-compatible format (`ASN1_STRFLGS_RFC2253`).

    /* Complete set of RFC2253 flags */

    #define XN_FLAG_RFC2253 (ASN1_STRFLGS_RFC2253 | \
                             XN_FLAG_SEP_COMMA_PLUS | \
                             XN_FLAG_DN_REV | \
                             XN_FLAG_FN_SN | \
                             XN_FLAG_DUMP_UNKNOWN_FIELDS)

    /* readable oneline form */

    #define XN_FLAG_ONELINE (ASN1_STRFLGS_RFC2253 | \
                             ASN1_STRFLGS_ESC_QUOTE | \
                             XN_FLAG_SEP_CPLUS_SPC | \
                             XN_FLAG_SPC_EQ | \
                             XN_FLAG_FN_SN)

@mattklein123 @myidpt Do you guys have any preference for the format (examples in my previous comment)?